packages/fast-figma-plugin-msft/src/core/ui/index.tsx

```diff
@@ -0,0 +1,344 @@
+            action: MessageAction.assign,
+        });
+    };
+
+    private renderThemeSwitcher(): JSX.Element {
+        const key: keyof DesignSystem = "baseLayerLuminance";
+        const nodes = this.props.selectedNodes.filter(node =>
+            node.supports.includes("designSystem")
+        );
+        const themeData = nodes.map(node => node.designSystem.baseLayerLuminance);
+        const themesApplied = themeData.filter(value => typeof value === "number");
+        const nodeIds = nodes.map(node => node.id);
+
+        const style = {
+            marginInlineEnd: "12px",
+        };
+
+        const setLightTheme = this.setDesignSystemProperty.bind(
+            this,
+            key,
+            StandardLuminance.LightMode,
+            nodeIds
+        );
+
+        const setDarkTheme = this.setDesignSystemProperty.bind(
+            this,
+            key,
+            StandardLuminance.DarkMode,
+            nodeIds
+        );
+
+        const removeTheme = this.removeDesignSystemProperty.bind(this, key, nodeIds);
+
+        return (
+            <div style={{ marginTop: "4px" }}>
+                <Checkbox
+                    inputId={"theme-toggle"}
+                    checked={themesApplied.length > 0}
+                    onChange={themesApplied.length ? removeTheme : setLightTheme}
+                    style={style}
+                >
+                    <Label slot="label" htmlFor={"theme-toggle"}>
+                        Theme
+                    </Label>
+                </Checkbox>
+                <Radio
+                    inputId={"light-theme"}
+                    checked={themesApplied.includes(StandardLuminance.LightMode)}
+                    disabled={themesApplied.length === 0}
+                    name="theme"
+                    style={style}
+                    onChange={setLightTheme}
+                >
+                    <Label slot="label" htmlFor={"light-theme"}>
+                        Light
+                    </Label>
+                </Radio>
+                <Radio
+                    inputId={"dark-theme"}
+                    checked={themesApplied.includes(StandardLuminance.DarkMode)}
+                    disabled={themesApplied.length === 0}
+                    name="theme"
+                    onChange={setDarkTheme}
+                >
+                    <Label slot="label" htmlFor={"dark-theme"}>
+                        Dark
+                    </Label>
+                </Radio>
+            </div>
+        );
+    }
+
+    private setDesignSystemProperty<K extends keyof DesignSystem>(
+        property: K,
+        value: DesignSystem[K],
+        nodeIds: string[]
+    ): void {
+        this.props.dispatch({
+            type: MessageTypes.designSystem,
+            action: MessageAction.assign,
+            property,
+            value,
+            nodeIds,
+        });
+    }
+
+    private removeDesignSystemProperty<K extends keyof DesignSystem>(

```

Comment:
you used "delete" on an earlier method performing a similar action rather than remove, just wondering if you wanted to make those consistent.